src/main/java/gov/cabinetoffice/gap/adminbackend/services/ApplicationFormSectionService.java

```diff
@@ -118,4 +118,38 @@ public void updateSectionStatus(final Integer applicationId, final String sectio
         this.applicationFormRepository.save(applicationForm);
     }
 
+    public void updateSectionOrder(final Integer applicationId, final String sectionId, final Boolean upOrDown) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+
+        ApplicationFormEntity applicationForm = this.applicationFormRepository.findById(applicationId)
+                .orElseThrow(() -> new NotFoundException("Application with id " + applicationId + " does not exist"));
+
+        if (!session.getGrantAdminId().equals(applicationForm.getCreatedBy())) {
+            throw new AccessDeniedException("User " + session.getGrantAdminId()
+                    + " is unable to access the application form with id " + applicationId);
+        }
+
+        List<ApplicationFormSectionDTO> sections = applicationForm.getDefinition().getSections();
+        ApplicationFormSectionDTO section = applicationForm.getDefinition().getSectionById(sectionId);
+        int index = sections.indexOf(section);
+        int indexToSwap = upOrDown ? index - 1 : index + 1;
+
+        if (upOrDown) {
+            if (index <= 2) {
```

Comment:
Any hardcoded random number like this can be very confusing to people not familiar with the codebase. Would suggest something like:

```
    public boolean isMovedSectionIndexOutOfBounds(final int newIndex, final int sectionsSize) {
        final int CUSTOM_SECTION_STARTING_INDEX = 2;
        return newIndex <= CUSTOM_SECTION_STARTING_INDEX || newIndex >= sectionsSize - 1;
    }

Reply:
Fixed in latest commit